conf/broker.conf

```diff
@@ -892,10 +892,11 @@ managedLedgerCursorBackloggedThreshold=1000
 managedLedgerDefaultMarkDeleteRateLimit=1.0
 
 # Max number of entries to append to a ledger before triggering a rollover
-# A ledger rollover is triggered on these conditions
-#  * Either the max rollover time has been reached
-#  * or max entries have been written to the ledger and at least min-time
-#    has passed
+# A ledger rollover is triggered after the min rollover time has passed
+# and one of the following conditions is true:
```

Comment:
double check: `and` or `or`?

`and` means it must meet two conditions (for example, `after the min rollover time has passed` and `The max rollover time has been reached`), then a ledger rollover is triggered.

Reply:
I believe `and` is correct.

As I wrote in the description, I wrote this documentation assuming the logic in the code is correct. If we want to change the code, this documentation will need to change, too.

Here is my understanding of the Java. The first check is to evaluate the following: `spaceQuotaReached || maxLedgerTimeReached`. If it is false, it the ledger is not rolled. If it is true, the next step is to evaluate `config.getMinimumRolloverTimeMs() > 0`. If it is false, the method returns true and the ledger is rolled. If it returns true, it proceeds to return the result of `timeSinceLedgerCreationMs > config.getMinimumRolloverTimeMs()`. As such, the ledger will only be rolled once the min rollover time has passed. Technically, if you misconfigure pulsar, you could set the min rollover time to greater than the max rollover time, and the ledger would not rollover until the min rollover time passed. Once the min rollover time has passed, any of the three conditions for `spaceQuotaReached || maxLedgerTimeReached` (note that `spaceQuotaReached` is composed of two conditions) will cause the ledger to rollover.

Reply:
@michaeljmarshall thanks for your detailed explanations. 
@congbobo184 could you please help review?

Reply:
@Anonymitaet that's correct to me.